powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/CallbackUrlBehavior.java

```diff
@@ -431,7 +431,7 @@ private synchronized RestClient getRestClient(CallbackUrlEntity callbackUrlEntit
     private RestClient initializeRestClient(CallbackUrlEntity callbackUrlEntity) throws RestClientException {
         final DefaultRestClient.Builder builder = DefaultRestClient.builder();
         if (configuration.getHttpConnectionTimeout() != null) {
-            builder.connectionTimeout(configuration.getHttpConnectionTimeout());
+            builder.connectionTimeout(Math.toIntExact(configuration.getHttpConnectionTimeout().toMillis()));
```

Comment:
🤔 I have opened another issue, what we may improve in the next step. https://github.com/wultra/lime-java-core/issues/201